pageserver/src/storage_sync/upload.rs

```diff
@@ -231,6 +207,35 @@ where
     }
 }
 
+async fn upload_storage_object<P, S>(
```

Comment:
This is a bit asymmetric wit `download_storage_object`. It would be nice if this also took GenericRemoteStorage as argument, so that the callers didn't need to use match.